net/ip.cpp

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2023 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <cmath>
+#include <cstdint>
+#include <string.h>
```

Comment:
Prefer the C++ header variants. `<cstring>` instead of `<string.h>`. Also make sure you prefix things from the stdlib with `std::` as there's no guarantee that the implementation dumps things into the global namespace when using the C++ headers.